### ISSUE TITLE
Issues/can't compile

### ISSUE THREAD

**Issue by Welnnys** [david-legend/nimbus#7]
Hey there, I have been getting some errors when I try to compile/run it using "flutter run"

Launching lib\main.dart on Edge in debug mode...
../../../AppData/Local/Pub/Cache/hosted/pub.dev/google_fonts-2.1.0/lib/src/google_fonts_base.dart:15:1: Error: 'AssetManifest' is imported from both
'package:flutter/src/services/asset_manifest.dart' and 'package:google_fonts/src/asset_manifest.dart'.
import 'asset_manifest.dart';
^^^^^^^^^^^^^
../../../AppData/Local/Pub/Cache/hosted/pub.dev/auto_route-3.2.2/lib/src/router/provider/auto_route_information_provider.dart:30:47: Error: The required named parameter 'type' in method
'AutoRouteInformationProvider.routerReportsNewRouteInformation' is not required in overridden method 'RouteInformationProvider.routerReportsNewRouteInformation'.
      {required RouteInformationReportingType type}) {
                                              ^
/C:/src/flutter/packages/flutter/lib/src/widgets/router.dart:1440:8: Context: This is the overridden method ('routerReportsNewRouteInformation').
  void routerReportsNewRouteInformation(RouteInformation routeInformation, {RouteInformationReportingType type = RouteInformationReportingType.none}) {}
       ^
lib/app_theme.dart:29:5: Error: No named parameter with the name 'primaryVariant'.
    primaryVariant: AppColors.secondaryColor,
    ^^^^^^^^^^^^^^
/C:/src/flutter/packages/flutter/lib/src/material/color_scheme.dart:87:9: Context: Found this candidate, but the arguments don't match.
  const ColorScheme({
        ^^^^^^^^^^^
lib/app_theme.dart:21:7: Error: No named parameter with the name 'accentColor'.
      accentColor: colorScheme.primary,
      ^^^^^^^^^^^
/C:/src/flutter/packages/flutter/lib/src/material/theme_data.dart:237:11: Context: Found this candidate, but the arguments don't match.
  factory ThemeData({
          ^
../../../AppData/Local/Pub/Cache/hosted/pub.dev/google_fonts-2.1.0/lib/src/google_fonts_base.dart:33:31: Error: 'AssetManifest' is imported from both
'package:flutter/src/services/asset_manifest.dart' and 'package:google_fonts/src/asset_manifest.dart'.
AssetManifest assetManifest = AssetManifest();
                              ^^^^^^^^^^^^^
lib/presentation/widgets/buttons/nimbus_button.dart:40:21: Error: The argument type 'BorderRadius?' can't be assigned to the parameter type 'BorderRadiusGeometry' because 'BorderRadius?' is        
nullable and 'BorderRadiusGeometry' isn't.
 - 'BorderRadius' is from 'package:flutter/src/painting/border_radius.dart' ('/C:/src/flutter/packages/flutter/lib/src/painting/border_radius.dart').
 - 'BorderRadiusGeometry' is from 'package:flutter/src/painting/border_radius.dart' ('/C:/src/flutter/packages/flutter/lib/src/painting/border_radius.dart').
      borderRadius: borderRadius,
                    ^
../../../AppData/Local/Pub/Cache/hosted/pub.dev/gap-2.0.0/lib/src/rendering/gap.dart:84:29: Error: The getter '_direction' isn't defined for the class 'RenderGap'.
 - 'RenderGap' is from 'package:gap/src/rendering/gap.dart' ('../../../AppData/Local/Pub/Cache/hosted/pub.dev/gap-2.0.0/lib/src/rendering/gap.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named '_direction'.
    final Axis? direction = _direction;
                            ^^^^^^^^^^
../../../AppData/Local/Pub/Cache/hosted/pub.dev/gap-2.0.0/lib/src/rendering/gap.dart:98:29: Error: The getter '_direction' isn't defined for the class 'RenderGap'.
 - 'RenderGap' is from 'package:gap/src/rendering/gap.dart' ('../../../AppData/Local/Pub/Cache/hosted/pub.dev/gap-2.0.0/lib/src/rendering/gap.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named '_direction'.
    final Axis? direction = _direction;
                            ^^^^^^^^^^
Waiting for connection from debug service on Edge...               28.6s
Failed to compile application.

any idea how I can fix those?

Best wishes
Welnnys

**Comment by david-legend** [comment]
Yeah, it uses an old flutter version and you have a newer flutter version. Here are some options

1. The quickest way is to use the fvm to install the appropriate flutter version(v 2.12). That will get it to run
2. Another option is to migrate the entire website to use the flutter version you're on.

Let me know how it goes

**Comment by Welnnys** [comment]
Yep, I am using the latest flutter version. I can't seem to find a "stable" windows sdk with the version 2.12 on 

- https://docs.flutter.dev/release/archive?tab=windows

are you sure it is 2.12 or am I supposed to use one of the 2.12.* beta channel ones?

also for the second option by "migrate" you mean to recode the parts which are for the older version or is there a way I can migrate it using a command or something?

**Comment by david-legend** [comment]
use versions within 2.2.0 upwards, one of them should work.

No the tool I know helps migrate to null safety.
Its actually easier to do by hand if you're familiar with the new api

**Comment by Welnnys** [comment]
Sadly I am not familiar with the new api, I am new to flutter and wanted to use this template as a platform and recode most of it to my needs. I suppose you are not going to update the template to the newest flutter version (new api) any time soon?

**Comment by Welnnys** [comment]
Tried every 2.2.* version and I get the same error:

PS C:\Users\Trade\OneDrive\Работен плот\nimbus-main> flutter pub upgrade
Resolving dependencies...
Because every version of flutter from sdk depends on meta 1.3.0 and auto_route >=3.2.1 <4.1.0 depends on meta ^1.7.0, flutter from sdk is incompatible with auto_route >=3.2.1 <4.1.0.
So, because nimbus depends on both flutter any from sdk and auto_route ^3.2.2, version solving failed.
pub finished with exit code 1
PS C:\Users\Trade\OneDrive\Работен плот\nimbus-main> flutter run
Because every version of flutter_test from sdk depends on meta 1.3.0 and auto_route >=3.2.1 <4.1.0 depends on meta ^1.7.0, flutter_test from sdk is incompatible with auto_route >=3.2.1 <4.1.0.
So, because nimbus depends on both auto_route ^3.2.2 and flutter_test any from sdk, version solving failed.
Running "flutter pub get" in nimbus-main...                             
pub get failed (1; So, because nimbus depends on both auto_route ^3.2.2 and flutter_test any from sdk, version solving failed.)
PS C:\Users\Trade\OneDrive\Работен плот\nimbus-main> 
 *  History restored 

PS C:\Users\Trade\OneDrive\Работен плот\nimbus-main> flutter --version
Flutter 2.2.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b22742018b (2 years, 9 months ago) • 2021-05-14 19:12:57 -0700
Engine • revision a9d88a4d18
Tools • Dart 2.13.0
PS C:\Users\Trade\OneDrive\Работен плот\nimbus-main> 

I get this including the 3k Problems

**Comment by david-legend** [comment]
any luck ?
Do you want me to take a look ?

**Comment by Welnnys** [comment]
Everything is fine, I fixed it days ago :3

**Comment by Thamina33** [comment]
@Welnnys  which version u used

**Comment by david-legend** [comment]
so I should go ahead and merge ??

**Comment by tinypoppa** [comment]
I migrated mine successfully. Ifn you still need help, let me know.
